pkg/config/configuration.go

```diff
@@ -206,3 +276,272 @@ func containsKey(s []string, key string) bool {
+
+		// Iterate over all the operations and create a map for fast lookup
+		for _, appPolicy := range appPolicySpec.AppOperationActions {
+			// The operation name might be specified as /invoke/*
+			// Store the prefix as the key and use the remainder as post fix for faster lookups
+			// Also, prepend "/" in case it is missing in the operation name
+			operation := appPolicy.Operation
+			if !strings.HasPrefix(operation, "/") {
+				operation = "/" + operation
+			}
+			operationPrefix, operationPostfix := getOperationPrefixAndPostfix(operation)
+
+			operationActions := AccessControlListOperationAction{
+				OperationPostFix: operationPostfix,
+				VerbAction:       make(map[string]string),
+			}
+
+			// Iterate over all the http verbs and create a map and set the action for fast lookup
+			for _, verb := range appPolicy.HTTPVerb {
+				operationActions.VerbAction[verb] = appPolicy.Action
+			}
+
+			operationPolicy[operationPrefix] = operationActions
+		}
+		aclPolicySpec := AccessControlListPolicySpec{
+			AppName:             appPolicySpec.AppName,
+			DefaultAction:       appPolicySpec.DefaultAction,
+			TrustDomain:         appPolicySpec.TrustDomain,
+			Namespace:           appPolicySpec.Namespace,
+			AppOperationActions: operationPolicy,
+		}
+		accessControlList.PolicySpec[aclPolicySpec.AppName] = aclPolicySpec
+	}
+
+	if len(invalidTrustDomain) > 0 || len(invalidNamespace) > 0 || invalidAppName {
+		err := fmt.Errorf("invalid access control spec. %s, %s %s",
+			fmt.Sprintf("missing trustdomain for apps: %v", invalidTrustDomain),
+			fmt.Sprintf("missing namespace for apps: %v", invalidNamespace),
+			fmt.Sprintf("missing app name on at least one of the app policies: %v", invalidAppName))
+		return accessControlList, err
+	}
+
+	return accessControlList, nil
+}
+
+// GetAndParseSpiffeID retrieves the SPIFFE Id from the cert and parses it
+func GetAndParseSpiffeID(ctx context.Context) (*SpiffeID, error) {
+	spiffeID, err := getSpiffeID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := parseSpiffeID(spiffeID)
+	return id, err
+}
+
+func parseSpiffeID(spiffeID string) (*SpiffeID, error) {
+	if spiffeID == "" {
+		err := fmt.Errorf("input spiffe id string is empty")
+		return nil, err
+	}
+
+	if !strings.HasPrefix(spiffeID, SpiffeIDPrefix) {
+		err := fmt.Errorf("input spiffe id: %s is invalid", spiffeID)
+		return nil, err
+	}
+
+	// The SPIFFE Id will be of the format: spiffe://<trust-domain/ns/<namespace>/<app-id>
+	parts := strings.Split(spiffeID, "/")
+	if len(parts) < 6 {
+		err := fmt.Errorf("input spiffe id: %s is invalid", spiffeID)
+		return nil, err
+	}
+
+	var id SpiffeID
+	id.TrustDomain = parts[2]
+	id.Namespace = parts[4]
+	id.AppID = parts[5]
+
+	return &id, nil
+}
+
+func getSpiffeID(ctx context.Context) (string, error) {
+	var spiffeID string
+	peer, ok := peer.FromContext(ctx)
+	if ok {
+		if peer == nil || peer.AuthInfo == nil {
+			err := fmt.Errorf("unable to retrieve peer auth info")
+			return "", err
```

Comment:
please use `errors.New` instead of fmt.Errorf

return "", error.New("unable to retrieve peer auth info")